src/msgpack/core.clj

```diff
@@ -260,16 +264,16 @@
             ; str format family
             (= (bit-and 2r11100000 byte) 2r10100000)
             (let [n (bit-and 2r11111 byte)]
-              (String. ^bytes (read-bytes n data-input)))
+              (String. ^bytes (read-bytes n data-input) default-charset))
 
             (= byte 0xd9)
-            (String. ^bytes (read-bytes (read-uint8 data-input) data-input))
+            (String. ^bytes (read-bytes (read-uint8 data-input) data-input) default-charset)
 
             (= byte 0xda)
-            (String. ^bytes (read-bytes (read-uint16 data-input) data-input))
+            (String. ^bytes (read-bytes (read-uint16 data-input) data-input) default-charset)
```

Comment:
Can we define a small `read-string` helper method here, to avoid repeating ourselves? The existing code has a fair amount of repetition so let's fix that.

```clojure
(defn- read-string
  [n ^java.io.DataInput data-input]
  (String. ^bytes (read-bytes n data-input) msgpack-charset))
```